src/protocols/PhysicalLayer/PhysicalLayer.h

```diff
@@ -346,28 +360,29 @@ class PhysicalLayer {
     virtual RadioLibTime_t getTimeOnAir(size_t len);
 
     /*!
-      \brief Calculate the timeout value for this specific module / series (in number of symbols or units of time)
-      \param timeoutUs Timeout in microseconds to listen for
-      \returns Timeout value in a unit that is specific for the used module
+      \brief Calculate the timeout value for this specific module / series 
+      (in number of symbols or units of time).
+      \param timeoutUs Timeout in microseconds to listen for.
+      \returns Timeout value in a unit that is specific for the used module.
     */
     virtual RadioLibTime_t calculateRxTimeout(RadioLibTime_t timeoutUs);
 
     /*!
-      \brief Create the flags that make up RxDone and RxTimeout used for receiving downlinks
-      \param irqFlags The flags for which IRQs must be triggered
-      \param irqMask Mask indicating which IRQ triggers a DIO
+      \brief Create the flags that make up RxDone and RxTimeout used for receiving downlinks.
+      \param irqFlags The flags for which IRQs must be triggered.
+      \param irqMask Mask indicating which IRQ triggers a DIO.
       \returns \ref status_codes
     */
     virtual int16_t irqRxDoneRxTimeout(uint32_t &irqFlags, uint32_t &irqMask);
 
     /*!
-      \brief Check whether the IRQ bit for RxTimeout is set
-      \returns Whether RxTimeout IRQ is set
+      \brief Check whether a specific IRQ bit is set (e.g. RxTimeout, CadDone).
+      \returns Whether requested IRQ is set.
     */
-    virtual bool isRxTimeout();
+    virtual bool isIrqSet(uint8_t irq);
```

Comment:
I'm not sure it is sufficient to return boolean. There are three different outcomes for any specific IRQ check:

1. The IRQ is set - returns true
2. The IRQ is not set - returns false
3. The IRQ is not supported by the radio driver - returns false

Right now, it is not possible to distinguish whether the radio IRQ is not set, or unsupported.

The simplest solution I can think of is to return `int16_t` value of 0 if IRQ is not set, 1 if it is set and `RADIOLIB_ERR_UNSUPPORTED` if it does not exist.

Alternatively we would need to document which IRQs are supported for each of the radio drivers, which seems a bit tedious.

Reply:
Slightly inconvenient that negative values also turn into `true` when turned into a `bool`, but I understand and am OK with the solution.

Reply:
Then it might be a good idea to change the name of the method, e.g. `irqCheck`, and add macros for `RADIOLIB_IRQ_SET (1)` and `RADIOLIB_IRQ_UNSET (0)`, just to slightly discourage something like `if(isIrqSet(irq))`?

Reply:
Sorry, I don't really get what you mean?

Reply:
I meant that the name of the method `PhysicalLayer::isIrqSet()` could lead to some confusion - if you expect it to return boolean based on that name, you could do something like `if(radio.isIrqSet(RADIOLIB_IRQ_LR_FHSS_HOP))` and always have it evaluate as `true` if `radio` does not implement LR-FHSS. Personally, when I see a function named `ìsSomethingHappenning()` I do expect it to only return true or false, so that is a good point.

The name of the method could be changed to `PhysicalLayer::irqCheck()`, to not include that implicit "true/false" in the name itself.